toolkit/tools/pkg/imagecustomizerlib/bootcustomizer_test.go

```diff
@@ -0,0 +1,204 @@
+`
+	checkDiffs30(t, b, "", expectedDefaultGrubFileDiff)
+
+	err = b.UpdateSELinuxCommandLine(imagecustomizerapi.SELinuxModeForceEnforcing)
+	assert.NoError(t, err)
+
+	selinuxMode, err = b.getSELinuxModeFromGrub()
+	assert.NoError(t, err)
+	assert.Equal(t, imagecustomizerapi.SELinuxModeForceEnforcing, selinuxMode)
+
+	expectedDefaultGrubFileDiff = `5c5
+< GRUB_CMDLINE_LINUX="      rd.auto=1 net.ifnames=0 lockdown=integrity "
+---
+> GRUB_CMDLINE_LINUX="      rd.auto=1 net.ifnames=0 lockdown=integrity   security=selinux selinux=1 enforcing=1 "
+`
+	checkDiffs30(t, b, "", expectedDefaultGrubFileDiff)
+
+	err = b.UpdateSELinuxCommandLine(imagecustomizerapi.SELinuxModeDisabled)
+	assert.NoError(t, err)
+
+	selinuxMode, err = b.getSELinuxModeFromGrub()
+	assert.NoError(t, err)
+	assert.Equal(t, imagecustomizerapi.SELinuxModeDisabled, selinuxMode)
+
+	expectedDefaultGrubFileDiff = `5c5
+< GRUB_CMDLINE_LINUX="      rd.auto=1 net.ifnames=0 lockdown=integrity "
+---
+> GRUB_CMDLINE_LINUX="      rd.auto=1 net.ifnames=0 lockdown=integrity      "
+`
+	checkDiffs30(t, b, "", expectedDefaultGrubFileDiff)
+}
+
+func checkDiffs20(t *testing.T, b *BootCustomizer, expectedGrubCfgDiff string, expectedDefaultGrubFileDiff string) {
+	checkDiffs(t, b, filepath.Join(testDir, sampleGrubCfg20Path), filepath.Join(testDir, sampleDefaultGrub20Path),
+		expectedGrubCfgDiff, expectedDefaultGrubFileDiff)
+}
+
+func checkDiffs30(t *testing.T, b *BootCustomizer, expectedGrubCfgDiff string, expectedDefaultGrubFileDiff string) {
+	checkDiffs(t, b, filepath.Join(testDir, sampleGrubCfg30Path), filepath.Join(testDir, sampleDefaultGrub30Path),
+		expectedGrubCfgDiff, expectedDefaultGrubFileDiff)
+}
+
+func checkDiffs(t *testing.T, b *BootCustomizer, originalGrubCfgPath string, originalDefaultGrubFilePath string,
+	expectedGrubCfgDiff string, expectedDefaultGrubFileDiff string,
+) {
+	grubCfgDiff := calcDiff(t, originalGrubCfgPath, b.grubCfgContent)
+	defaultGrubFileDiff := calcDiff(t, originalDefaultGrubFilePath, b.defaultGrubFileContent)
+
+	assert.Equal(t, expectedGrubCfgDiff, grubCfgDiff, "diff of grub.cfg file")
+	assert.Equal(t, expectedDefaultGrubFileDiff, defaultGrubFileDiff, "diff of /etc/default/grub file")
+}
+
+func calcDiff(t *testing.T, oldPath string, newContent string) string {
+	diff, _, err := shell.ExecuteWithStdin(newContent, "diff", oldPath, "-")
+	if err != nil {
+		exitError, isExitError := err.(*exec.ExitError)
+		if assert.True(t, isExitError) {
+			assert.Equal(t, exitError.ExitCode(), 1)
+		}
+	}
+	return diff
+}
+
+func createBootCustomizerFor20(t *testing.T) *BootCustomizer {
+	return createBootCustomizer(t, filepath.Join(testDir, sampleGrubCfg20Path),
+		filepath.Join(testDir, sampleDefaultGrub20Path), false)
+}
+
+func createBootCustomizerFor30(t *testing.T) *BootCustomizer {
+	return createBootCustomizer(t, filepath.Join(testDir, sampleGrubCfg30Path),
+		filepath.Join(testDir, sampleDefaultGrub30Path), true)
+}
```

Comment:
Why the separate cases for 2.0 and 3.0? At the moment we're not guaranteeing cross-version backwards compatibility with the toolkit as far as I know. Is MIC's policy different?

Reply:
Yes. MIC supports 2.0 and 3.0 with the code from the 3.0 branch.